contracts/integrations/test/utils/function_assertions.ts

```diff
@@ -58,11 +70,75 @@ export class FunctionAssertion {
         }
 
         // Call the after condition.
-        const afterInfo = await this.condition.after(beforeInfo, callResult, ...args);
+        const afterInfo =
+            this.condition.after !== undefined
+                ? await this.condition.after(beforeInfo, callResult, ...args)
+                : undefined;
 
         return {
             beforeInfo,
             afterInfo,
         };
     }
 }
+
+/**
+ * Note: This can be treated like a normal `FunctionAssertion`.
+ */
+export class FunctionAssertionSequence {
+    public readonly assertions: Assertion[] = [];
+    public readonly inputGenerators: Array<() => any[]> = [];
+
+    /**
+     * Set up a set of function assertions equipped with generator functions.
+     * A number can be specified for each assertion that will detail the frequency
+     * that this assertion will be called relative to the other assertions in the
+     * set.
+     */
+    constructor(assertionGenerators: [Assertion, (() => any[])][]) {
+        for (const assertionGenerator of assertionGenerators) {
+            this.assertions.push(assertionGenerator[0]);
+            this.inputGenerators.push(assertionGenerator[1]);
+        }
+    }
+
+    /**
+     * Run the functions in this assertion set in order.
+     */
+    public async runAsync(environment: BlockchainTestsEnvironment): Promise<void> {
+        for (let i = 0; i < this.assertions.length; i++) {
+            await this.assertions[i].runAsync(...this.inputGenerators[i]());
+        }
+    }
+}
+
+export class ContinuousFunctionAssertionSet {
```

Comment:
I kinda liked just `FunctionAssertionSet` better tbh

Reply:
Interesting. The reason I changed this is because `FunctionAssertionSequence` terminates in a finite number of steps, whereas `ContinuousFunctionAssertionSet` will not (and will instead run continuously until the process is killed) . I was thinking that there could be a continuous variant of `FunctionAssertionSequence` and a finite version of `ContinuousFunctionAssertionSet`. With this said, I think that a more descriptive name is warranted, although maybe this isn't the right name. 

Thoughts on a better name?

Reply:
o I see; we can leave it as is then